libcodechecker/report.py

```diff
@@ -151,6 +151,51 @@ def compare_ctrl_sections(curr, prev):
         return ''
 
 
+def generate_report_hash_no_bugpath(main_section, source_file):
```

Comment:
I think it would be better to create one function for this with multiple function parameter which could skip elements from report hash generation such as checker name, bug step, etc.:
```py
def generate_report_hash(main_section, source_file, skip_checker_name=False, skip_bug_step=False)
```
This way we could reduce code duplications. What do you think @gyorb?

Reply:
I think it is more visible this way. It is much easier to use the wrong hash in the code, if only the function arguments are different.